src/views/forms/FormPersistentDelegationRequestTransaction/FormPersistentDelegationRequestTransactionTs.ts

```diff
@@ -381,20 +367,45 @@ export class FormPersistentDelegationRequestTransactionTs extends FormTransactio
             txs.push(nodeUnLinkTx);
         }
 
-        if (this.action !== HarvestingAction.STOP) {
-            if (!this.isAccountKeyLinked || !this.currentSignerHarvestingModel?.encRemotePrivateKey) {
+        if (this.action === HarvestingAction.START) {
+            if (this.isAccountKeyLinked && !this.remoteAccountPrivateKey) {
+                const accountUnlinkTx = this.createAccountKeyLinkTx(
+                    this.currentSignerAccountInfo.supplementalPublicKeys?.linked.publicKey,
+                    LinkAction.Unlink,
+                    maxFee,
+                );
+                txs.push(accountUnlinkTx);
+            }
+            if (this.isVrfKeyLinked && !this.vrfPrivateKey) {
+                const vrfUnlinkTx = this.createVrfKeyLinkTx(
+                    this.currentSignerAccountInfo.supplementalPublicKeys?.vrf.publicKey,
+                    LinkAction.Unlink,
+                    maxFee,
+                );
+                txs.push(vrfUnlinkTx);
+            }
+            if (!this.isAccountKeyLinked || !this.remoteAccountPrivateKey) {
                 const accountKeyLinkTx = this.createAccountKeyLinkTx(this.newRemoteAccount.publicKey, LinkAction.Link, maxFee);
                 txs.push(accountKeyLinkTx);
             }
-            if (!this.isVrfKeyLinked || !this.currentSignerHarvestingModel?.encVrfPrivateKey) {
+            if (!this.isVrfKeyLinked || !this.vrfPrivateKey) {
                 const vrfKeyLinkTx = this.createVrfKeyLinkTx(this.newVrfKeyAccount.publicKey, LinkAction.Link, maxFee);
                 txs.push(vrfKeyLinkTx);
             }
 
-            if (!this.isNodeKeyLinked) {
-                const nodeLinkTx = this.createNodeKeyLinkTx(this.formItems.nodeModel.nodePublicKey, LinkAction.Link, maxFee);
-                txs.push(nodeLinkTx);
-            }
+            // link node key
+            const nodeLinkTx = this.createNodeKeyLinkTx(this.formItems.nodeModel.nodePublicKey, LinkAction.Link, maxFee);
+            txs.push(nodeLinkTx);
```

Comment:
while this is required for delegated harvesting, it is not required for non-delegated harvesting. 
whereas the other two links (remote, vrf) are required for all forms of harvesting.
unsure if this should be mandatory?

also, unlike the others, it appears this link is not cleared? is that intentional?

Reply:
> while this is required for delegated harvesting, it is not required for non-delegated harvesting.

this form is handling `delegated-harvesting` only, `non-delegating` harvesters (node-owners) can only monitor their state here.

> also, unlike the others, it appears this link is not cleared? is that intentional?

if it's already linked, it's always being unlinked first(on line 361). 
so if linked regardless of the action (STOP or START)  => unlink the current node key link
and if the action is START, additionally => link the new node key